src/levanter/data/audio.py

```diff
@@ -271,6 +273,9 @@ def __iter__(self):
             del singleton_dict["audio_shape"]
             yield singleton_dict
 
+    def seek(self):
```

Comment:
wdyt about seek just taking an int that tells you where to seek?

Reply:
That's where I started, but the hard part there is what that int should represent and how that changes at different levels of the code. 

At the trainer level, we want to seek to the batch corresponding to a training step, but at lower levels we need to seek to different levels of granularity (e.g. some datasets need to seek to the right start token, some need to seek to the right document, and ultimately all of them need to seek to the right chunk in storage).

I think that if we use `seek(index: int)` as the arg format, it might get a bit confusing what that integer represents at different parts of the code base. I think that also risks `seek` logic diverging from the actual data loading logic which could lead to subtle resume bugs.

Since the actual data loading logic uses generators, using generators in the seek code made it feels easier to write seek in such a way that I could largely keep it joined to the existing data loading logic while making the perf improvements really close to the expensive file reads.

Reply:
It could be that I just couldn't see the path to write a clean version of the `seek(index: int)`, but it felt like I'd end up having to do arithmetic at each level to convert from the index that made sense at one level to the index that was needed at the lower level, rather than just piggy pack on the data loading logic directly.

Reply:
i think you do have to do that arithmetic, but maybe it's not worth. i have an idea that is going to be better overall than everything we have now. I need to whiteboard it tomorrow